apps/genetics/src/ot-ui-components/helpers/downloadTable.ts

```diff
@@ -1,25 +1,33 @@
-import FileSaver from 'file-saver';
+import * as FileSaver from 'file-saver';
+
+export type TableRow = { [key: string]: any };
+export type HeaderMap = {
+  id: string;
+  label: string;
+  export?: (o: Object) => any;
+}[];
+export type DownloadFormat = 'csv' | 'tsv' | 'json';
 
 const UNEXPECTED_FORMAT =
   'Unexpected format. Supported options are csv, tsv and json.';
 
-const pick = (object, keys) => {
+const pick = (object: TableRow, keys: HeaderMap) => {
   return keys.reduce(function (o, k) {
     // take into account optional export() function, which takes precedence as per other download formats
     o[k.id] = k.export ? k.export(object) : object[k.id];
     return o;
-  }, {});
+  }, {} as TableRow);
 };
 
-const quoteIfString = d => (typeof d === 'string' ? `"${d}"` : d);
+const quoteIfString = (d: any) => (typeof d === 'string' ? `"${d}"` : d);
 
-const asJSONString = ({ rows, headerMap }) => {
+const asJSONString = (rows: TableRow[], headerMap: HeaderMap) => {
   // use the full headerMap which contain optional export() function for each header
   const rowsHeadersOnly = rows.map(row => pick(row, headerMap));
   return JSON.stringify(rowsHeadersOnly);
 };
 
-const asCSVString = ({ rows, headerMap }) => {
+const asCSVString = (rows: TableRow[], headerMap: HeaderMap) => {
```

Comment:
Please keep the params of the exporting function as objects, this is a good practice cause help to avoid the necessity to respect order in parameters

Reply:
Understood, done!